osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -276,6 +276,13 @@ private void selectNextSet(int direction, bool skipDifficulties)
 
         private void selectNextDifficulty(int direction)
         {
+            if (selectedBeatmapSet == null || selectedBeatmap == null)
```

Comment:
The beatmap check feels a little gratuitous. `CarouselBeatmapSet` should ensure that at least one child is selected, and trying to create a beatmapset without difficulties results in it just not being added to carousel.